Python/gc.c

```diff
@@ -1313,7 +1314,7 @@ gc_collect_young(PyThreadState *tstate,
     if (scale_factor < 1) {
         scale_factor = 1;
     }
-    gcstate->work_to_do += survivor_count + survivor_count * SCAN_RATE_MULTIPLIER / scale_factor;
+    gcstate->work_to_do += gcstate->heap_size / SCAN_RATE_DIVISOR / scale_factor;
```

Comment:
IIUC, this is the core of the fix.  Is that right?

And just to be sure I understood, this has the concrete effect of making GC collection happen more frequently.  Is that right?

Without having more than cursory knowledge about the GC implementation, the change in frequency seems like a significant change relative to the performance profile of GC (which is the point of the fix, of course).  The relative effect is clearly derived from the overhead of collection vs. the cost of larger collections (and the continual cost of larger generations).  Are there other tradeoffs to which we need to pay attention?

Reply:
Yes, this is the important change.

It may make GC happen more frequently, but is more likely to make it happen *less* frequently.
The important thing is that the long term rate of incremental GC is the same as long term rate of the old generation collection prior to https://github.com/python/cpython/pull/116206.